rotkehlchen/chain/evm/decoding/interfaces.py

```diff
@@ -437,15 +437,16 @@ def __init__(
         self.save_data_to_cache_method = save_data_to_cache_method
         self.read_data_from_cache_method = read_data_from_cache_method
         self.chain_id = chain_id
-        if self.chain_id is None:
-            self.cache_data = self.read_data_from_cache_method()
-        else:
-            self.cache_data = self.read_data_from_cache_method(chain_id)
+        self.cache_data: dict[int, Any] = {}
```

Comment:
So you changed all `self.cache_data` to be a dict of int to whatever cache is at that index entry? Why dict? Are you trying to achieve an ordered set? Why not just a list?

Reply:
I used dict because now in all the places where we use it, we can call .get(0, {}). This way we avoid the if checks. Also, the reload_data() in mixin was assuming the correctness of the length of self.cache_data list, because it was using zip() with strict=True. And since we want this mixin to be used in both cases with len(cache_data) = 1 and 2, the dict makes that flexible.